packages/Notification/Notification.jsx

```diff
@@ -31,32 +34,111 @@ const renderIcon = icon => <DecorativeIcon symbol={icon.symbol} variant={icon.co
  *
  * @version ./package.json
  */
-const Notification = ({ variant, children, ...rest }) => (
-  <Box {...safeRest(rest)} vertical={3} dangerouslyAddClassName={styles[variant]}>
-    <FlexGrid>
-      <FlexGrid.Row>
-        <FlexGrid.Col>
-          <Box inline between={3}>
-            {isImportant(variant) ? renderIcon(iconByVariant[variant]) : undefined}
-            <div
-              className={joinClassNames(
-                isImportant(variant) ? styles.withIcon : styles.withoutIcon
+class Notification extends React.Component {
+  state = {
+    dismissed: false,
+    contentWrapperHeight: undefined,
+  }
+
+  componentDidMount() {
+    if (this.props.dismissible) {
+      this.adjustContentHeight()
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.props.dismissible) {
+      this.adjustContentHeight()
+    }
+  }
+
+  adjustContentHeight = () => {
+    if (this.contentWrapper.offsetHeight !== this.state.contentWrapperHeight) {
+      this.setState({ contentWrapperHeight: this.contentWrapper.offsetHeight })
+    }
+  }
+
+  renderNotification() {
+    const { variant, dismissible, dismissibleA11yLabel, children, ...rest } = this.props
+
+    return (
+      <Box
+        {...safeRest(rest)}
+        inset={3}
+        dangerouslyAddClassName={joinClassNames(styles.base, styles[variant])}
+      >
+        <FlexGrid>
+          <FlexGrid.Row>
+            <FlexGrid.Col xs={dismissible ? 11 : undefined}>
+              <Box inline between={3}>
+                {isImportant(variant) ? renderIcon(iconByVariant[variant]) : undefined}
+                <div
+                  className={joinClassNames(
+                    isImportant(variant) ? styles.withIcon : styles.withoutIcon
+                  )}
+                >
+                  <Paragraph bold={isImportant(variant)}>{children}</Paragraph>
+                </div>
+              </Box>
+            </FlexGrid.Col>
+          </FlexGrid.Row>
+        </FlexGrid>
+        {dismissible && (
```

Comment:
I'm assuming here: would the dismissible icon be displaced to the far right on desktop viewports? Maybe it's more appropriate to add another `<FlexGrid.Col xs={dismissible ? 1 : 0}>` that wraps the `<StandaloneIcon>`.

Reply:
I didn't include the icon in the FlexGrid because the icon is required to be 8px from top and 16px from the right. I think it's better to just absolutely position it there.